### PR TITLE
Make namespace configurable for resources in capabilites package

### LINKED ISSUE #2256: Make namespace configurable for resources in capabilities package
**Describe the feature request**
Right now the resources in capabilities packages are only installed on `tkg-system` namespace, we want to provide an ability for the person who installs the package to configure where the package resources should be created..

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by yharish991**
### What this PR does / why we need it

This PR makes the namespace configurable for resources in the capabilities package.

This PR also additionally
1. fixes the vendir.yml location in tkr-service package
2. vendirs capabilities CRD

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/2256

### Describe testing done for PR

1. Generated capabilities package bundle(PACKAGE_NAME=capabilities OCI_REGISTRY=${REGISTRY} PACKAGE_REPOSITORY=standalone make package-bundle)
2. Installed the capabilities package on k8s cluster by passing the namespace value in a secret used by PackageInstall CR

### Release note

```release-note
Made namespace configurable for resources in capabilities package
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: packages/standalone/capabilities/vendir.lock.yml [hunk 1]
```diff
@@ -1,11 +1,15 @@
 apiVersion: vendir.k14s.io/v1alpha1
 directories:
 - contents:
-  - manual: {}
+  - git:
+      commitTitle: Updating version tag used for BOM retrieval to v0.20.0 (#1951)...
+      sha: 66368ce04abcb92be48d439b09b39722a45a727b
```
Comment (codegold79) [inline packages/standalone/capabilities/vendir.lock.yml:6]: Not blocking. Just a question: does this sha get updated every time a new version of Tanzu Framework is released? Or is it updated when this component is upgraded?

**Review by codegold79 — CHANGES_REQUESTED** [review]

File: packages/standalone/capabilities/vendir.lock.yml (hunk 1 above)
Comment (yharish991) [reply packages/standalone/capabilities/vendir.lock.yml:6]: we need to manually update the sha in [vendir.yml](https://github.com/vmware-tanzu/tanzu-framework/pull/2257/files#diff-bf10adcc40972f9e0890ed9bdd733d4f0d1d08b883baf03491e512ffec9a20c8R10) whenever we make a change to capabilities CRD, right now I pulled the sha of the latest tanzu-framework release commit

**Review by codegold79 — APPROVED** [review]

File: packages/standalone/capabilities/bundle/config/values.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #@data/values
 ---
 
+namespace:
```
Comment (danniel1205) [inline packages/standalone/capabilities/bundle/config/values.yaml:4]: question: do we want to provide a default value for ns ? Wondering if ytt will assign `namespace: null` to the Deployment.metadata.namespace if nothing is provided.

Comment (yharish991) [reply packages/standalone/capabilities/bundle/config/values.yaml:4]: i don't have any preference, we left the default namespace empty for other packages too.. resolving this comment to keep this package consistent with other packages in the repo

**Review by danniel1205 — APPROVED** [review]
LGTM. Add an approval on behalf of core-reviewers

**Pull request merged**